File: sample-videochat-webrtc/sample-videochat-webrtc/Components/Users/Views/PaddingLabel/PaddingLabel.m
//
//  PaddingLabel.m
//  sample-videochat-webrtc
//
//  Created by Injoit on 04.08.2021.
//  Copyright © 2021 QuickBlox Team. All rights reserved.
//

#import "PaddingLabel.h"

@interface PaddingLabel()
//MARK: - Properties
@property (assign, nonatomic) UIEdgeInsets textPaddingInsets;

@end

@implementation PaddingLabel
//MARK: - Life Cycle
- (instancetype)init {
    self = [super init];
    if (self) {
        _textPaddingInsets = UIEdgeInsetsZero;
    }
    return self;
}

- (CGRect)textRectForBounds:(CGRect)bounds limitedToNumberOfLines:(NSInteger)numberOfLines {
    CGRect textRect = [super textRectForBounds:bounds limitedToNumberOfLines:numberOfLines];
    UIEdgeInsets invertedInsets = UIEdgeInsetsMake(-self.textPaddingInsets.top,
                                                   -self.textPaddingInsets.left,
                                                   -self.textPaddingInsets.bottom,
                                                   -self.textPaddingInsets.right);
    return UIEdgeInsetsInsetRect(textRect, invertedInsets);
}

- (void)drawTextInRect:(CGRect)rect {
    [super drawTextInRect:UIEdgeInsetsInsetRect(rect, self.textPaddingInsets)];
}

//MARK: - Public Methods
- (void)setupTextPaddingInsets:(UIEdgeInsets)textPaddingInsets {
    _textPaddingInsets = textPaddingInsets;
    [self invalidateIntrinsicContentSize];
}

@end

File: sample-videochat-webrtc-swift/sample-videochat-webrtc-swift/Components/Users/SelectedUsersCountAlert/PaddingLabel.swift
//
//  PaddingLabel.swift
//  sample-videochat-webrtc-swift
//
//  Created by Injoit on 30.07.2021.
//  Copyright © 2021 QuickBlox. All rights reserved.
//

import UIKit

class PaddingLabel: UILabel {
    //MARK: - Properties
    var textPaddingInsets = UIEdgeInsets.zero {
         didSet { invalidateIntrinsicContentSize() }
     }
     
    //MARK: - Life Cycle
     override func textRect(forBounds bounds: CGRect, limitedToNumberOfLines numberOfLines: Int) -> CGRect {
         let textRect = super.textRect(forBounds: bounds, limitedToNumberOfLines: numberOfLines)
         let invertedInsets = UIEdgeInsets(top: -textPaddingInsets.top,
                                           left: -textPaddingInsets.left,
                                           bottom: -textPaddingInsets.bottom,
                                           right: -textPaddingInsets.right)
         return textRect.inset(by: invertedInsets)
     }
     
     override func drawText(in rect: CGRect) {
         super.drawText(in: rect.inset(by: textPaddingInsets))
     }
}
